src/core/io/writer/h5md/h5md_core.cpp

```diff
@@ -0,0 +1,707 @@
+/*
+  Copyright (C) 2010,2011,2012,2013,2014,2015,2016 The ESPResSo project
+  Copyright (C) 2002,2003,2004,2005,2006,2007,2008,2009,2010 
+    Max-Planck-Institute for Polymer Research, Theory Group
+  
+  This file is part of ESPResSo.
+  
+  ESPResSo is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+  
+  ESPResSo is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+  
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>. 
+*/
+
+
+#include "h5md_core.hpp"
+
+
+namespace writer {
+namespace h5md {
+
+
+/* Constructor for the File class. */
+File::File(std::string const &filename, std::string const &script_name)
+{
+    boost::filesystem::path script_path(script_name);
+    this->absolute_script_path = boost::filesystem::canonical(script_path);
+    /* Store the filename in a member variable. */
+    this->user_filename = filename;
+    /* Get number of local particles. */
+    this->n_local_part = cells_get_n_particles();
+    if(!(this->n_local_part > 0)) {
+        throw std::runtime_error("Please first set up particles before initializing the H5md object.");
+    } 
+    /* Check if a file with given filename exists. */
+    bool file_exists = this->check_file_exists(filename);
+    /* If it exists, check for the H5MD structure. */
+    if (file_exists)
+    {
+        this->has_H5MD_structure = check_for_H5MD_structure(filename);
+    }
+    if (file_exists && this->has_H5MD_structure)
+    {
+        /*
+         * If the file exists and has a valid H5MD structure, lets create a new file and copy the hdf5 objects.
+         * This has the advantage, that the new file can just be deleted if the simulation crashes at some point and we
+         * still have a valid trajectory, we can start from.
+        */
+        std::string temp_name = filename + "_tmp";
+        this->lastfile = h5xx::file(filename, MPI_COMM_WORLD, MPI_INFO_NULL,
+                                     h5xx::file::out);
+        this->h5md_file = h5xx::file(temp_name, MPI_COMM_WORLD, MPI_INFO_NULL,
+                                     h5xx::file::out);
+        /* particles -- atoms -- box -- edges */
+        H5Ocopy(lastfile.hid(), "/particles", this->h5md_file.hid(), "/particles", H5P_DEFAULT,
```

Comment:
Don't you store time series? So wouldn't this allow you to refer to another files for earlier time steps? Or is the problem that there is not one object per time step and data field, but one object per data field and each time step is appended to that same object?